examples/mugen/data/mugen_datamodules.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Callable, NamedTuple, Optional
+
+import pytorch_lightning as pl
+import torch
+import torch.distributed as dist
+import torch.utils.data as data
+
+from .mugen_dataset import MUGENDataset, MUGENDatasetArgs
+
+
+class MUGENDataModuleArgs(NamedTuple):
+    batch_size: int = 16
+    num_workers: int = 4
+
+
+class MUGENDataModule(pl.LightningDataModule):
+    """General lightning data module for MUGEN dataset.
+
+    Args:
+        mugen_dataset_args (MUGENDatasetArgs): arguments for MUGENDataset.
+        datamodule_args (MUGENDataModuleArgs): arguments for this LightningDataModule.
+            See MUGENDataModuleArgs definition for defaults.
+        text_transform (Optional[Callable]): transform for text batches.
+            Only used when not ``None`` and when ``mugen_dataset_args.get_text_desc = True``.
+            Defaults to ``None``.
+        video_transform (Optional[Callable]): transform for video batches.
+            Only used when not ``None`` and when ``mugen_dataset_args.get_game_frame = True``.
+            Defaults to ``None``.
+        audio_transform (Optional[Callable]): transform for audio batches.
+            Only used when not ``None`` and when ``mugen_dataset_args.get_audio = True``.
+            Defaults to ``None``.
+        shuffle (bool): whether to reshuffle data after each epoch.
+            Defaults to ``True``.
+    """
+
+    def __init__(
+        self,
+        mugen_dataset_args: MUGENDatasetArgs,
+        data_module_args: MUGENDataModuleArgs = MUGENDataModuleArgs(),
+        text_transform: Optional[Callable] = None,
+        video_transform: Optional[Callable] = None,
+        audio_transform: Optional[Callable] = None,
+        shuffle=True,
```

Comment:
nit: we don't run type checking on examples but since you defined the type for every other argument might as well add it for `shuffle` as well